flycheck.el

```diff
@@ -6241,6 +6242,39 @@ See URL `https://github.com/kisielk/errcheck'."
     ;; packages, and can't check individual Go files.
     (and (flycheck-buffer-saved-p) (flycheck-go-package-name))))
 
+(defun flycheck-groovy-write-compiler ()
+  "Write the compilation program to temporary file so it can be executed."
+  (let* ((dir (flycheck-temp-dir-system))
+         (file (expand-file-name "compiler.groovy" dir))
+         (prog "import org.codehaus.groovy.control.*
+
+file = new File(args[0])
+unit = new CompilationUnit()
+unit.addSource(file)
+
+try {
+    unit.compile(Phases.CONVERSION)
+} catch (MultipleCompilationErrorsException e) {
+    e.errorCollector.write(new PrintWriter(System.out, true), null)
+}
+"))
+
+    (write-region prog nil file nil 'no-message)
+    file))
```

Comment:
I don't like this.  We write this file onto the disk for every single syntax check?  Can't we give it as argument to `groovy`, i.e. like we do for Emacs Lisp with `--eval`?